src/Endpoint/EndpointBase.php

```diff
@@ -30,19 +30,18 @@ abstract class EndpointBase
     /**
      * @param array<string, mixed> $configuration
      */
-    public function __construct(protected RestInterface $rest, string $baseUrl, array $configuration = [])
+    public function __construct(protected RestInterface $rest, array $configuration = [])
     {
-        $configuration['endpointUrl'] = rtrim($baseUrl, '/') . static::ENDPOINT_URL_PATH;
         $this->configuration = $this->getConfigurationResolver()->resolve($configuration);
     }
 
     protected function getConfigurationResolver(): OptionsResolver
     {
         $resolver = new OptionsResolver();
 
-        $resolver->setRequired('endpointUrl')
```

Comment:
Validating only `endpointUrl` is a bug, it causes problems if the `baseUrl` is empty.